python/ray/air/util/data_batch_conversion.py

```diff
@@ -218,7 +218,7 @@ def _cast_ndarray_columns_to_tensor_extension(df: pd.DataFrame) -> pd.DataFrame:
         for col_name, col in df.items():
             if column_needs_tensor_extension(col):
                 try:
-                    df.loc[:, col_name] = TensorArray(col)
+                    df[col_name] = TensorArray(col)
```

Comment:
We want to mutate (overwrite) the column in-place, so `df.loc[:, col_name] = TensorArray(col)` is preferred.